File: src/main/java/com/uitnetwork/alexa/model/AlexaIntentRequest.java
package com.uitnetwork.alexa.model;

import com.amazon.speech.json.SpeechletRequestEnvelope;
import com.amazon.speech.speechlet.IntentRequest;

public class AlexaIntentRequest {
    public final AlexaIntent alexaIntent;

    public final SpeechletRequestEnvelope<IntentRequest> intentRequestEnvelope;

    public AlexaIntentRequest(AlexaIntent alexaIntent, SpeechletRequestEnvelope<IntentRequest> intentRequestEnvelope) {
        this.alexaIntent = alexaIntent;
        this.intentRequestEnvelope = intentRequestEnvelope;
    }

    public AlexaIntent getAlexaIntent() {
        return alexaIntent;
    }

    public SpeechletRequestEnvelope<IntentRequest> getIntentRequestEnvelope() {
        return intentRequestEnvelope;
    }
}

File: src/main/kotlin/com/uitnetwork/alexa/model/AlexaIntentRequest.kt
package com.uitnetwork.alexa.model

import com.amazon.speech.json.SpeechletRequestEnvelope
import com.amazon.speech.speechlet.IntentRequest

data class AlexaIntentRequest(val alexaIntent: AlexaIntent,
                              val intentRequestEnvelope: SpeechletRequestEnvelope<IntentRequest>)
